test/lib/jdk/test/lib/cds/CDSTestUtils.java

```diff
@@ -599,10 +604,72 @@ public static File getSourceFile(String name) {
         return new File(dir, name);
     }
 
+    // Check commandline for the last instance of Xshare to see if the process can load
+    // a CDS archive
+    public static boolean maybeRunningWithArchive(List<String> cmd) {
+      // -Xshare only works for the java executable
+      if (!cmd.get(0).equals(JDKToolFinder.getJDKTool("java")) || cmd.size() < 2) {
+        return false;
+      }
+
+      Pattern share = Pattern.compile("-Xshare:.*");
+      String lastXShare = "";
+      for (String s : cmd) {
+        Matcher m = share.matcher(s);
+        if (m.find()) {
+          lastXShare = s;
+        }
+      }
+
+      if (lastXShare.equals("-Xshare:dump") || lastXShare.equals("-Xshare:off")) {
+        return false;
+      }
+      return true;
+    }
+
+    static boolean maybeAddOption(String cmd, boolean hasGCOption, ArrayList<String> cmdline) {
+      Pattern gc = Pattern.compile("^-XX:\\+.*GC");
```

Comment:
This could be a static variable so it can be used by different methods and only compiled once.